sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2080,7 +2080,7 @@ class Analyzer(override val catalogManager: CatalogManager)
             throw QueryCompilationErrors.expectPersistentFuncError(
               nameParts.head, cmd, mismatchHint, u)
           } else {
-            ResolvedNonPersistentFunc(nameParts.head, V1Function(info))
+            ResolvedNonPersistentFunc(nameParts.last, V1Function(info))
```

Comment:
is it a bug?

Reply:
I think it should be a bug.  should always be the `name`:
```
/**
 * A plan containing resolved non-persistent (temp or built-in) function.
 */
case class ResolvedNonPersistentFunc(
    name: String,
    func: UnboundFunction)
  extends LeafNodeWithoutStats {
  override def output: Seq[Attribute] = Nil
}
```

Reply:
can we have a separate PR to fix this bug?

Reply:
sorry, my change was confusing, it is not a bug:

```
        lookupBuiltinOrTempFunction(nameParts)
          .orElse(lookupBuiltinOrTempTableFunction(nameParts))
```
could be non-empty only when `nameparts.length == 1`,  so `nameParts.head` is fine.